chia/util/write_file.py

```diff
@@ -0,0 +1,24 @@
+import os
+import shutil
+
+from pathlib import Path
+from typing import Union
+
+
+def write_file(file_path: Path, data: Union[str, bytes], *, file_mode: int = 0o600):
+    """
+    Writes the provided data to a temporary file and then moves it to the final destination.
+    """
+
+    dir_perms: int = 0o700
+    # Create the parent directory if necessary
+    os.makedirs(os.path.dirname(file_path), mode=dir_perms, exist_ok=True)
+
+    temp_path: Path = file_path.with_suffix("." + str(os.getpid()))
+    mode: str = "w" if type(data) == str else "wb"
+    with open(os.open(str(temp_path), os.O_CREAT | os.O_EXCL | os.O_WRONLY, file_mode), mode) as f:
```

Comment:
I see, you're using `O_EXCL` here, which would throw an exception if the file already existed, right?
Do we just give up in that case and fail everything?

Reply:
Changed to use the tempfile module to guarantee a unique temp file for writing.